paddle/fluid/operators/controlflow/while_op.cc

```diff
@@ -87,14 +101,48 @@ class WhileOp : public framework::OperatorBase {
     bool is_test = Attr<bool>("is_test");
     auto &skip_vars = Attr<std::vector<std::string>>(kSkipEagerDeletionVars);
     VLOG(2) << GetSkipEagerDeletionVarsDebugString(skip_vars);
-
     auto ctx = executor.Prepare(*program, block->ID(), skip_vars);
     if (!is_test) {
       while (cond_data) {
         auto &current_scope = scope.NewScope();
         step_scopes->push_back(&current_scope);
+        // Each loop the value of LoDTensor in input would be saved sub_scope to
+        // ensure that the correct values can be used in the gradient
+        // calculation.
+        auto input_names = Inputs(kX);
+        for (auto &input_var_name : input_names) {
+          if (std::find(no_copy_var_names.begin(), no_copy_var_names.end(),
+                        input_var_name) == no_copy_var_names.end()) {
+            auto input_var_rename = input_var_name + "@RENAME";
```

Comment:
Because other places we used "@RENAME", I suggest you to use other string to make local code safe.

Reply:
Thanks, done.